torch/csrc/autograd/custom_function.h

```diff
@@ -11,4 +11,49 @@ TORCH_API variable_list _wrap_outputs(
   const std::unordered_set<at::TensorImpl*> &dirty_inputs,
   const at::ArrayRef<Variable> raw_outputs,
   const std::shared_ptr<Function> &cdata);
+
+// To use custom autograd operations implement a CFunction subclass with
+// static backward and forward functions
+//
+// forward() can take as many arguments as you want and should return a
+// variable list. Use of any direct Variable arguments will be registered in
+// the graph but no vectors/sets or any other data structures will be traversed.
+//
+// backward() will be given a variable list containing as many Variables as
+// there were outputs from forward. It should return as many Variables as there
+// were inputs with each of them containing the gradient w.r.t. its
+// corresponding input
+//
+// For example:
+// class MyFunction : public CFunction<MyFunction> {
+//   public:
+//   static variable_list forward(AutogradContext *ctx, int n, Variable var);
+//
+//   static variable_list backward(AutogradContext *ctx, variable_list grad_output);
+// };
+// To use MyFunction
+// Variable x;
+// MyFunction::apply(6, x);
+
+template <class T>
+struct CFunction {
+  template<typename... Args>
+  static variable_list apply(Args&&... args);
+};
+
+struct AutogradContext {
```

Comment:
Is there any way in way to allow the user to store arbitrary data in the ctx between the forward and the backward the same way it can be done in python?
Even if it just subclassing this, it should be in the example.

Reply:
Yes, I have added another pr for AutogradContext : https://github.com/pytorch/pytorch/pull/23184